### PR TITLE
WIP: Entity Generator 

### PR DESCRIPTION
**Pull request by deepu105**
From https://github.com/jhipster/jhipster-kotlin/pull/36

@sendilkumarn @hberndt FYI

### CODE REVIEW

**Comment by deepu105** [comment]
@hberndt can you test https://github.com/hipster-labs/generator-jhipster-sample-blueprint and see if the entity generation works here?

I'm trying to figure out what is causing issue here since both these have similar code and for me the sample-blueprint works but not the kotlin one

**Comment by deepu105** [comment]
@sendilkumarn @hberndt the issue causing the original error is fixed in jhipster master, now the blueprint works but you still have some errors in your files.js due to wrong path etc, please take care of those

File: generators/entity-server/templates/src/test/kotlin/package/repository/search/EntitySearchRepositoryMockConfiguration.java.ejs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<%#
```
Comment (sendilkumarn) [inline generators/entity-server/templates/src/test/kotlin/package/repository/search/EntitySearchRepositoryMockConfiguration.java.ejs]: rename this file to `.kt`

File: generators/entity-server/files.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+const constants = require('generator-jhipster/generators/generator-constants');
+
+/* Constants use throughout */
+const INTERPOLATE_REGEX = constants.INTERPOLATE_REGEX;
+const SERVER_MAIN_RES_DIR = constants.SERVER_MAIN_RES_DIR;
+const TEST_DIR = constants.TEST_DIR;
+const SERVER_MAIN_SRC_KOTLIN_DIR = `${constants.MAIN_DIR}kotlin/`;
+const SERVER_TEST_SRC_KOTLIN_DIR = `${constants.TEST_DIR}kotlin/`;
+
+// const BASE_DIR = '../../../node_modules/generator-jhipster/generators/entity-server/templates/';
```
Comment (sendilkumarn) [inline generators/entity-server/files.js]: change this into > `const BASE_DIR = '../../../../../../node_modules/generator-jhipster/generators/entity-server/templates';`

File: generators/entity-server/files.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+ * The default is to use a file path string. It implies use of the template method.
+ * For any other config an object { file:.., method:.., template:.. } can be used
+ */
+const serverFiles = {
+    db: [
+        {
+            condition: generator => generator.databaseType === 'sql',
+            path: SERVER_MAIN_RES_DIR,
+            templates: [{
+                file: 'config/liquibase/changelog/added_entity.xml',
```
Comment (sendilkumarn) [inline generators/entity-server/files.js]: use ``` 
file: `${BASE_DIR}/src/main/resources/config/liquibase/changelog/added_entity.xml`,```

**Pull request merged**